### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by lerfast**
Hello fellow code reviewer, I did the following tasks:

In this PR, I have added validations to the Blog App models and written unit tests to ensure the functionality and integrity of the models.

Changes:
Validations:

User Model:
Ensured that the name is present.
Ensured that posts_counter is a number and is greater than or equal to zero.
Post Model:
Ensured that the title is present and does not exceed 250 characters.
Ensured that comments_counter is a number and is greater than or equal to zero.
Ensured that likes_counter is a number and is greater than or equal to zero.
RSpec Testing:

Added unit tests for the methods and validations in the User, Post, Comment, and Like models.
Verified that all tests pass without any failures.
Manual Testing:

Performed manual testing in the Rails console to validate the functionality of the models and their methods.

### CODE REVIEW

File: spec/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+RSpec.describe User, type: :model do
+  it 'is not valid without a name' do
+    user = User.new(name: nil)
+    expect(user).to_not be_valid
+  end
+
+  it 'is not valid with a negative posts_counter' do
+    user = User.new(posts_counter: -1)
+    expect(user).to_not be_valid
+  end
```
Comment (Lucas-Masaba) [inline spec/user_spec.rb:12]: - [ ] Any custom method in your models should have corresponding tests. For example, the `three_most_recent_posts` method in your User model should've tested

File: spec/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+RSpec.describe Post, type: :model do
+  it 'is not valid without a title' do
+    post = Post.new(title: nil)
+    expect(post).to_not be_valid
+  end
+
+  it 'is not valid with a title longer than 250 characters' do
+    post = Post.new(title: 'a' * 251)
+    expect(post).to_not be_valid
+  end
```
Comment (Lucas-Masaba) [inline spec/post_spec.rb:12]: - [ ] Any custom method in your models should have corresponding tests. For example, the `update_posts_counter` and `recent_comments` methods in your Post model should be tested

File: spec/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  it 'increments the comments counter when created' do
+    expect { Comment.create(text: 'Test Comment', user:, post:) }.to change {
+                                                                       post.reload.comments_counter
+                                                                     }.by(1)
+  end
+
+  it 'decrements the comments counter when destroyed' do
+    comment = Comment.create(text: 'Test Comment', user:, post:)
+    expect { comment.destroy }.to change { post.reload.comments_counter }.by(-1)
+  end
```
Comment (Lucas-Masaba) [inline spec/comment_spec.rb:16]: - [ ] Any custom method in your models should have corresponding tests. For example, the `update_comments_counter` method in your Post model should be tested

File: spec/like_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  let!(:post) { Post.create!(title: 'Test Post', author: user, comments_counter: 0, likes_counter: 0) }
+
+  it 'increments the likes counter when created' do
+    expect { Like.create(user:, post:) }.to change { post.reload.likes_counter }.by(1)
+  end
+
+  it 'decrements the likes counter when destroyed' do
+    like = Like.create(user:, post:)
+    expect { like.destroy }.to change { post.reload.likes_counter }.by(-1)
+  end
```
Comment (Lucas-Masaba) [inline spec/like_spec.rb:14]: - [ ] Any custom method in your models should have corresponding tests. For example, the `update_likes_counter` method in your Post model should be tested

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @lerfast  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- All written tests passing :heavy_check_mark: 
- Successfully added validations :gear: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/user_spec.rb (hunk 1 above)
Comment (lerfast) [reply spec/user_spec.rb:12]: image
did it!!

File: spec/post_spec.rb (hunk 2 above)
Comment (lerfast) [reply spec/post_spec.rb:12]: image
done!!!

File: spec/comment_spec.rb (hunk 3 above)
Comment (lerfast) [reply spec/comment_spec.rb:16]: image
Done!!!

File: spec/like_spec.rb (hunk 4 above)
Comment (lerfast) [reply spec/like_spec.rb:14]: image
Did it!!!

**Comment by lerfast** [comment]
image

**Review by OyePriscilla — APPROVED** [review]
Hi @lerfast 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
 


## Highlights
Implemented test for custom methods ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**